bob-backend/src/pearl/holder.rs

```diff
@@ -79,9 +79,14 @@ impl Holder {
         storage.blobs_count().await
     }
 
-    pub async fn index_memory(&self) -> usize {
+    pub async fn active_index_memory(&self) -> usize {
         let storage = self.storage.read().await;
-        storage.index_memory().await
+        storage.active_index_memory().await
+    }
+
+    pub async fn freeable_resources_memory(&self) -> usize {
```

Comment:
As I see, we only need to know the fact, that holder has some freeable resources (the size of occupied memory is not needed). Thus I suggest to replace this function with `have_excess_resources` that returns `bool`